### PR TITLE
data science for review

### PR DESCRIPTION
**Pull request by gguntupalli**

### CODE REVIEW

File: data_science/parser.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+require_relative 'sample'
+
+class Parser
+  def self.parse(file)
+    if File.exist?(file)
+      data = JSON.parse(File.read(file))
+      data.map { |row| Sample.new(row) }
+    else
+      raise 'File not found'
+    end
```
Comment (zsyed1) [inline data_science/parser.rb]: Lines 6-11 could be written using a guard condition:
```
raise 'File not found' unless File.exist?(file)
data = JSON.parse(File.read(file))
data.map { |row| Sample.new(row) }
```

File: data_science/spec/sample_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require_relative 'spec_helper'
+
+describe Sample do
+  sample = {}
+  sample['date'] = '2014-03-20'
+  sample['cohort'] = 'B'
+  sample['result'] = 0
+
```
Comment (zsyed1) [inline data_science/spec/sample_spec.rb]: For lines 4 - 8 you could build the hash like:
```
sample = { 'data' => '2014-03-20', 'cohort' => 'B', 'result' => 0 }
```

Comment (zsyed1) [inline data_science/spec/sample_spec.rb]: Actually it should be a `let`:
```
let(:sample) { {'data' ....} }
```

File: data_science/spec/sample_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      expect(sample_object.date).to eq(sample['date'])
+    end
+    it 'should have cohort' do
+      expect(sample_object.cohort).to eq(sample['cohort'])
+    end
+    it 'should have result' do
+      expect(sample_object.result).to eq(sample['result'])
+    end
+  end
+end
```
Comment (zsyed1) [inline data_science/spec/sample_spec.rb]: Nice work! This looks done to me.

File: data_science/calculator.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def ab_analyzer_data
+    values = {}
+    @data.uniq(&:cohort).each do |sample|
+      values[sample.cohort] ||= {
+        success: conversions(sample.cohort),
+        failure: (sample_size(sample.cohort) - conversions(sample.cohort)) }
+    end
+    values
```
Comment (djkotowski) [inline data_science/calculator.rb]: You can skip L36 and L42 by using `#each_with_object` instead of `#each`.

File: data_science/calculator.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  def conversions(cohort)
+    @data.count { |sample| sample.cohort == cohort && sample.result == 1 }
+  end
+
+  def conversion_rate(cohort)
+    (conversions(cohort).to_f / sample_size(cohort)).round(DECIMAL_PLACES)
+  end
+
+  def standard_error(cohort)
+    p = conversion_rate(cohort)
```
Comment (djkotowski) [inline data_science/calculator.rb]: Avoid using single-letter variables, even if they match what the standard error formula uses.

File: data_science/calculator.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    @data.uniq(&:cohort).each do |sample|
+      values[sample.cohort] ||= {
+        success: conversions(sample.cohort),
+        failure: (sample_size(sample.cohort) - conversions(sample.cohort)) }
+    end
+    values
+  end
+
+  def confidence_level
+    tester = ABAnalyzer::ABTest.new ab_analyzer_data
```
Comment (djkotowski) [inline data_science/calculator.rb]: Wrap initializer params with parentheses.

File: data_science/calculator.rb (hunk 5 above)
Comment (zsyed1) [inline data_science/calculator.rb]: @djkotowski Disagree!!!

Comment (djkotowski) [inline data_science/calculator.rb]: @zsyed1: This is a nitpick, but I find that single-letter variables that are short for something else are hard to read.  This depends on the formula.

**Pull request closed without merging**